### PR TITLE
Improve tooling and sbt-plugin error handling

### PR DESCRIPTION
**Pull request by jonas**
Supersedes #142 (except for the `config` change).

### CODE REVIEW

File: tests/src/test/scala/org/scalanative/bindgen/BindgenSpec.scala [hunk 1]
```diff
@@ -18,15 +18,21 @@ class BindgenSpec extends FunSpec {
...
         .name(name)
         .link("bindgentests")
         .packageName("org.scalanative.bindgen.samples")
         .excludePrefix("__")
         .generate()
-        .writeToFile(outputFile)
+
+      result match {
+        case Right(binding) =>
+          binding.writeToFile(outputFile)
```
Comment (jonas) [inline tests/src/test/scala/org/scalanative/bindgen/BindgenSpec.scala]: My suggestion is to eventually remove error emitting code from these tests so we can add `assert(binding.errors.isEmpty)` here.

Comment (kornilova-l) [reply tests/src/test/scala/org/scalanative/bindgen/BindgenSpec.scala]: But we still want to check what bindings are generated if a header file cannot be perfectly converted to Scala code.
For example we check that literal define is skipped if its value does not fit into Scala type.
Even if we check in `BindgenReportingSpec` that warning is printed for this case it does not guarantee that the code is not outputted.

File: tests/src/test/scala/org/scalanative/bindgen/BindgenReportingSpec.scala [hunk 2]
```diff
@@ -33,51 +33,57 @@ class BindgenReportingSpec extends FunSpec {
           .excludePrefix("__")
           .generate()
 
+        result match {
+          case Right(binding)      => assert(binding.errors == errors)
+          case Left(bindingErrors) => assert(bindingErrors == errors)
```
[the remark this answers is not in the document]
Comment (jonas) [reply tests/src/test/scala/org/scalanative/bindgen/BindgenReportingSpec.scala]: Good idea, done

File: tools/src/main/scala/org/scalanative/bindgen/Bindgen.scala [hunk 3]
```diff
@@ -52,7 +53,7 @@ sealed trait Bindgen {
   /**
    * Run binding generator
    */
-  def generate(): Bindings
+  def generate(): Either[Seq[String], Bindings]
```
[the remark this answers is not in the document]
Comment (jonas) [reply tools/src/main/scala/org/scalanative/bindgen/Bindgen.scala]: True, thanks. Done.

File: tests/src/test/scala/org/scalanative/bindgen/BindgenSpec.scala (hunk 1 above)
Comment (jonas) [reply tests/src/test/scala/org/scalanative/bindgen/BindgenSpec.scala]: Assuming these are edge cases it could still make sense to have them in the reporting spec.
